graylog2-server/src/main/java/org/graylog2/rest/resources/system/SessionsResource.java

```diff
@@ -126,10 +127,5 @@ public SessionResponse newSession(@Context ContainerRequestContext requestContex
     public void terminateSession(@ApiParam(name = "sessionId", required = true) @PathParam("sessionId") String sessionId) {
         final Subject subject = getSubject();
         securityManager.logout(subject);
-
-        final org.apache.shiro.session.Session session = subject.getSession(false);
```

Comment:
At this place the check does not make sense and returned a 404 every time, that's why it was removed.